commit dcf16b6c1b2614d32a46d0deacf53fc1d3adcadb
Author: David Faure <faure@kde.org>
Date:   Fri Oct 7 21:36:52 2011 +0200

    Fix crash when the collection list from the collectionfetch job is empty.
    
    Reported (and fix tested) by arkascha on irc.

diff --git a/agents/maildispatcher/sendjob.cpp b/agents/maildispatcher/sendjob.cpp
index 2d59a2b..52b0e27 100644
--- a/agents/maildispatcher/sendjob.cpp
+++ b/agents/maildispatcher/sendjob.cpp
@@ -301,15 +301,20 @@ void SendJob::Private::doPostJob( bool transportSuccess, const QString &transpor
 void SendJob::Private::slotSentMailCollectionFetched(KJob* job)
 {
   Akonadi::Collection fetchCol;
-  if( job->error() ) {
+  bool ok = false;
+  if( !job->error() ) {
+    const CollectionFetchJob *const fetchJob = qobject_cast<CollectionFetchJob*>( job );
+    if ( !fetchJob->collections().isEmpty() ) {
+        fetchCol = fetchJob->collections().first();
+        ok = true;
+    }
+  }
+  if ( !ok ) {
     if ( !SpecialMailCollections::self()->hasDefaultCollection( SpecialMailCollections::SentMail ) ) {
       abortPostJob();
       return;
     }
     fetchCol = SpecialMailCollections::self()->defaultCollection( SpecialMailCollections::SentMail );
-  } else {
-    const CollectionFetchJob *const fetchJob = qobject_cast<CollectionFetchJob*>( job );
-    fetchCol = fetchJob->collections().first();
   }
   currentJob = new ItemMoveJob( item, fetchCol, q );
   QObject::connect( currentJob, SIGNAL( result( KJob* ) ), q, SLOT( postJobResult( KJob* ) ) );
